bot/telegram_bot.py

```diff
@@ -590,30 +591,90 @@ async def handle_callback_inline_query(self, update: Update, context: CallbackCo
                         f'{localized_text("error", bot_language)}. '
                         f'{localized_text("try_again", bot_language)}'
                     )
-                    await self.edit_message_with_retry(context,
-                                                       chat_id=None,
-                                                       message_id=inline_message_id,
+                    await self.edit_message_with_retry(context, chat_id=None, message_id=inline_message_id,
                                                        text=f'{query}\n\n_{answer_tr}:_\n{error_message}',
                                                        is_inline=True)
                     return
 
-            # Edit the current message to indicate that the answer is being processed
-            await context.bot.edit_message_text(inline_message_id=inline_message_id,
-                                                text=f'{query}\n\n_{answer_tr}:_\n{loading_tr}',
-                                                parse_mode=constants.ParseMode.MARKDOWN)
+                if self.config['stream']:
+                    stream_response = self.openai.get_chat_response_stream(chat_id=user_id, query=query)
+                    i = 0
+                    prev = ''
+                    sent_message = None
+                    backoff = 0
+                    async for content, tokens in stream_response:
+                        if len(content.strip()) == 0:
+                            continue
 
-            logging.info(f'Generating response for inline query by {name}')
-            response, used_tokens = await self.openai.get_chat_response(chat_id=user_id, query=query)
-            self.add_chat_request_to_usage_tracker(user_id, used_tokens)
+                        cutoff = 180 if len(content) > 1000 else 120 if len(content) > 200 else 90 if len(
+                            content) > 50 else 50
+                        cutoff += backoff
+
+                        if i == 0:
+                            try:
+                                if sent_message is not None:
+                                    await self.edit_message_with_retry(context, chat_id=None,
+                                                                       message_id=inline_message_id,
+                                                                       text=f'{query}\n\n_{answer_tr}:_\n{content}',
+                                                                       is_inline=True)
+                            except:
+                                continue
```

Comment:
Why not log?

Reply:
@jvican good question. I have used the logic from the main prompt function, as the base for the inline query streaming.
I think we can add logging to both and also try to combine the semi-duplicated code 

https://github.com/n3d1117/chatgpt-telegram-bot/blob/756a3fed4f149f5ab8604782eb81893487e45ebd/bot/telegram_bot.py#L459

Reply:
+1 for future logic isolation! Yes we can add a log here no problem, I didn't do that initially because these operations (delete+send initial message) failed very rarely in my testing, and even if they did they will be just retried when the next chunk comes in